src/MultiSelect/MultiSelect.tsx

```diff
@@ -0,0 +1,235 @@
+import * as classNames from "classnames";
+import * as React from "react";
+import { useState } from "react";
+import { useCombobox, useMultipleSelection } from "downshift";
+import * as FontAwesome from "react-fontawesome";
+
+import Label from "../Label/Label";
+import { FormElementSize, formElementSizeClassName } from "../utils/Forms";
+import { Values } from "../utils/types";
+
+import "./MultiSelect.less";
+
+type Item = { key: string; value?: React.ReactNode; stringValue: string };
+
+export interface Props {
+  className?: string;
+  name: string;
+  label: string;
+  hideLabel?: boolean;
+  placeholder?: string;
+  items: Item[];
+  // list of keys
+  value: string[];
+  creatable?: boolean;
+  onChange?: (items: Item[]) => void;
+  size?: Values<typeof FormElementSize>;
+}
+
+export const cssClass = {
+  CONTAINER: "MultiSelect--container",
+  LABEL: "MultiSelect--label",
+  LABEL_HIDDEN: "MultiSelect--label--hidden",
+  SELECT_CONTAINER: "MultiSelect--selectContainer",
+  SELECT_CONTAINER_FOCUSED: "MultiSelect--selectContainer--focused",
+  SELECTED_ITEMS_CONTAINER: "MultiSelect--selectedItemsContainer",
+  SELECTED_ITEM_CONTAINER: "MultiSelect--selectedItemContainer",
+  SELECTED_ITEM_BUTTON: "MultiSelect--selectedItemButton",
+  INPUT: "MultiSelect--input",
+  CARET_BUTTON: "MultiSelect--caretButton",
+  MENU_RESET: "MultiSelect--items--reset",
+  MENU_OPEN: "MultiSelect--items--open",
+  MENU_ITEM: "MultiSelect--item",
+  MENU_ITEM_HIGHLIGHTED: "MultiSelect--item--highlighted",
+  NO_ITEMS_FOUND: "MultiSelect--itemsNoneFound",
+};
+
+function getSelectableItems(items: Item[], selectedItems: Item[], inputValue: string) {
+  return items.filter(
+    (item) =>
+      !selectedItems.some((si) => si.key === item.key) &&
+      (inputValue === "" ||
+        item.stringValue.toLocaleLowerCase().includes(inputValue.toLocaleLowerCase())),
+  );
```

Comment:
I think this could potentially be slow if the number of items is very large, and about half the items are selected. To avoid potential slowness in that case, you could create a set of `selectedItems.key`s at the top of this function.
Another small optimization is to store `inputValue.toLocaleLowerCase()` in a variable at the top of this function. 
I'm not sure how much this would come up in our use cases, so these optimizations may end up being unecessary.

Reply:
Nice! Sounds good.

I'm going to refactor the creatable code a little bit in a future PR, so I'll make those optimizations there